packages/indexer-common/src/indexer-management/monitor.ts

```diff
@@ -549,11 +555,113 @@ export class NetworkMonitor {
     return deployments
   }
 
+  async latestValidEpoch(network?: string): Promise<NetworkEpochBlock> {
+    network = network ?? this.blockOracleSubgraph.network
+    this.logger.trace(`latest valid epoch`, {
+      settlementNetwork: this.blockOracleSubgraph.network,
+      network,
+      networkAlias: NetworkAliases[network],
+    })
+    try {
+      const result = await this.blockOracleSubgraph.query(
+        gql`
+          query globalStates($network: String!) {
+            globalStates {
+              latestValidEpoch {
+                epochNumber
+                blockNumbers(where: { network: $network }) {
+                  network {
+                    id
+                  }
+                  blockNumber
+                  previousBlockNumber {
+                    blockNumber
+                  }
+                }
+              }
+            }
+          }
+        `,
+        {
+          network: NetworkAliases[network],
+        },
+      )
+
+      if (result.error) {
+        throw result.error
+      }
+
+      if (!result.data.globalStates) {
+        throw indexerError(
+          IndexerErrorCode.IE069,
+          `Failed to query latest valid epoch number for the global state`,
+        )
+      }
+
+      const epochBlock = await buildEpochBlock(
+        this.indexingStatusResolver,
+        result.data.globalStates[0].latestValidEpoch.epochNumber,
+        result.data.globalStates[0].latestValidEpoch.blockNumbers[0],
+      )
+
+      return epochBlock
+    } catch (err) {
+      this.logger.error(
+        `Failed to query latest global epoch number from epoch block oracle`,
+        {
+          err,
+        },
+      )
+      throw err
+    }
+  }
+
+  async fetchPOIBlockPointer(allocation: Allocation): Promise<BlockPointer> {
+    try {
+      const networkID = (
+        await this.indexingStatusResolver.indexingStatus([
+          allocation.subgraphDeployment.id,
+        ])
+      )[0].chains[0].network
+      const epoch = await this.latestValidEpoch(networkID)
+      const epochStartBlock = {
+        number: epoch.startBlockNumber,
+        hash: epoch.startBlockHash,
+      }
+      // // Previous epoch also acceptable by the arbitration charter, but not sure if we should leave this
+      // // as there could be delay between epoch update and EBO syncs
```

Comment:
I think we'll need to leave in the fallback to previous epoch in case the Epoch Block Oracle has delay in sending the message at start of epoch.

Reply:
I was thinking that during the EBO's delay period, this fallback would set the latest epoch to the previous of previous epoch and resolve POI from an invalid epoch. So what do you think if I move this fallback to be an error-catch in `resolvePOI` with the current epoch block? we can also only make queries to the previous block numbers when it is needed?

Reply:
I'm sorry but I don't think I'm following exactly.  

If the epoch block oracle subgraph does not yet have data for the current epoch, so effectively the agent cannot know what the start block for current epoch is then it should use the start block from epoch = currentEpoch - 1 to resolve the POI. 

What do you mean by?  
- "set the latest epoch to the previous of previous epoch and resolve POI from an invalid epoch"? 
- "move this fallback to be an error-catch in resolvePOI with the current epoch block"?

Reply:
Yes, that makes. Previous my understanding for the delay between new epoch starts and the EBO subgraph update is that the agent would receive old epoch data, and resolve POI with an epoch that's the old previous epoch. Now I understand that EBO subgraph simply won't return data, I also added the extra check with network epoch manager contract